### PR TITLE
feat(install script): macOS install script

### PR DESCRIPTION
**Pull request by cpheps**
### Proposed Change
- Added a macOS installation script that wraps the brew commands and handles install, upgrade, and uninstall.

#### Install

```sh
           888                                        8888888 .d88888b.
           888                                          888  d88P" "Y88b
           888                                          888  888     888
   .d88b.  88888b.  .d8888b   .d88b.  888d888 888  888  888  888     888
  d88""88b 888 "88b 88K      d8P  Y8b 888P"   888  888  888  888     888
  888  888 888  888 "Y8888b. 88888888 888     Y88  88P  888  888 Y8b 888
  Y88..88P 888 d88P      X88 Y8b.     888      Y8bd8P   888  Y88b.Y8b88P
   "Y88P"  88888P"   88888P'  "Y8888  888       Y88P  8888888 "Y888888"
                                                                   Y8b

===================================================
| Checking Prerequisites
===================================================
  Checking that the operating system is supported...
    Succeeded!
  Checking for script dependencies...
    Succeeded!
  Prerequisite check complete!

===================================================
| Checking for existing install
===================================================
  No current install

===================================================
| Installing observIQ OpenTelemetry Collector
===================================================
  Tapping formula...
    Succeeded!
  Installing collector...
Password:
    Succeeded!
  Enabling service...
    Succeeded!
  observIQ OpenTelemetry Collector installation complete!

===================================================
| Information
===================================================
  Collector Home:     /opt/homebrew/opt/observiq-otel-collector
  Collector Config:   /opt/homebrew/opt/observiq-otel-collector/config.yaml
  Start Command:      launchctl start com.observiq.collector
  Stop Command:       launchctl stop com.observiq.collector
  Logs Command:       tail -f /tmp/observiq-otel-collector.log

===================================================
| Support
===================================================
  For more information on configuring the collector, see the docs:
    https://github.com/observiq/observiq-otel-collector/tree/main#observiq-opentelemetry-collector
  If you have any other questions please contact us at support@observiq.com

===================================================
| Installation Complete!
===================================================
```

#### Upgrade

```sh
           888                                        8888888 .d88888b.
           888                                          888  d88P" "Y88b
           888                                          888  888     888
   .d88b.  88888b.  .d8888b   .d88b.  888d888 888  888  888  888     888
  d88""88b 888 "88b 88K      d8P  Y8b 888P"   888  888  888  888     888
  888  888 888  888 "Y8888b. 88888888 888     Y88  88P  888  888 Y8b 888
  Y88..88P 888 d88P      X88 Y8b.     888      Y8bd8P   888  Y88b.Y8b88P
   "Y88P"  88888P"   88888P'  "Y8888  888       Y88P  8888888 "Y888888"
                                                                   Y8b

===================================================
| Upgrading observIQ OpenTelemetry Collector
===================================================
  Checking if install exists ...
    Succeeded!
  Stopping service...
    Succeeded!
  Upgrading collector...
    Succeeded!
  Starting service...
    Succeeded!

===================================================
| Upgrade Complete!
===================================================
```

#### Uninstall

```sh
           888                                        8888888 .d88888b.
           888                                          888  d88P" "Y88b
           888                                          888  888     888
   .d88b.  88888b.  .d8888b   .d88b.  888d888 888  888  888  888     888
  d88""88b 888 "88b 88K      d8P  Y8b 888P"   888  888  888  888     888
  888  888 888  888 "Y8888b. 88888888 888     Y88  88P  888  888 Y8b 888
  Y88..88P 888 d88P      X88 Y8b.     888      Y8bd8P   888  Y88b.Y8b88P
   "Y88P"  88888P"   88888P'  "Y8888  888       Y88P  8888888 "Y888888"
                                                                   Y8b

===================================================
| Uninstalling observIQ OpenTelemetry Collector
===================================================
  Stopping service...
    Succeeded!
  Removing service...
    Succeeded!
  Uninstalling collector...
    Succeeded!
  Untapping formula...
    Succeeded!

===================================================
| Uninstallation Complete!
===================================================
```

##### Checklist
- [x] Changes are tested
- [ ] CI has passed

### CODE REVIEW

File: docs/installation-mac.md [hunk 1]
```diff
@@ -37,21 +47,19 @@ The collector uses `brew services` the following commands control the collector
...
 # Start the collector
-brew services start observiq/observiq-otel-collector/observiq-otel-collector
+launchctl start com.observiq.collector
 
 # Stop the collector
-brew services stop observiq/observiq-otel-collector/observiq-otel-collector
-
-# Restart the collector
-brew services restart observiq/observiq-otel-collector/observiq-otel-collector
+launchctl stop com.observiq.collector
```
Comment (BinaryFissionGames) [inline docs/installation-mac.md:53]: What's the context for this change to using launchctl start/stop for service control?

I've seen three different ways to start/stop services in Darwin (the previous brew commands, launchctl start/stop, and also launchctl load/unload, which is used by `kardianos/service` for starting and stopping services), so I just want to understand why launchctl start/stop is preferred here.

Comment (cpheps) [reply docs/installation-mac.md:53]: The brew services commands didn't work and I couldn't find any reason why. Might be something to revisit. I'm not super familiar with launchctl so I'll look into load/unload and see how that differs.

Comment (cpheps) [reply docs/installation-mac.md:53]: @BinaryFissionGames From the launchctl help it describes load and unload as:

```
load            Recommended alternatives: bootstrap | enable. Bootstraps a service or directory of services.
unload          Recommended alternatives: bootout | disable. Unloads a service or directory of services.
```

and start/stop:

```
start           Starts the specified service.
stop            Stops the specified service if it is running.
```

from this I think start/stop is correct as we don't want to bootstrap or unbootstrap the service but rather just turn it on/off.

Comment (BinaryFissionGames) [reply docs/installation-mac.md:53]: Cool, sounds good

**Review by BinaryFissionGames — APPROVED** [review]

**Review by jsirianni — APPROVED** [review]

**Pull request merged**